File: app/src/main/java/com/u1fukui/bbs/ui/main/MainNavigator.java
package com.u1fukui.bbs.ui.main;


import android.content.Intent;

import com.u1fukui.bbs.ui.creation.thread.CreateThreadActivity;
import com.u1fukui.bbs.ui.notification.NotificationListActivity;

import javax.inject.Inject;

public class MainNavigator extends ThreadListNavigator {

    @Inject
    public MainNavigator(MainActivity activity) {
        super(activity);
    }

    public void navigateToCreateThreadPage() {
        Intent intent = new Intent(getActivity(), CreateThreadActivity.class);
        getActivity().startActivity(intent);
    }

    public void navigateToNotificationPage() {
        Intent intent = new Intent(getActivity(), NotificationListActivity.class);
        getActivity().startActivity(intent);
    }
}

File: app/src/main/java/com/u1fukui/bbs/ui/main/MainNavigator.kt
package com.u1fukui.bbs.ui.main


import android.content.Intent

import com.u1fukui.bbs.ui.creation.thread.CreateThreadActivity
import com.u1fukui.bbs.ui.notification.NotificationListActivity

import javax.inject.Inject

class MainNavigator @Inject
constructor(activity: MainActivity) : ThreadListNavigator(activity) {

    fun navigateToCreateThreadPage() {
        val intent = Intent(activity, CreateThreadActivity::class.java)
        activity.startActivity(intent)
    }

    fun navigateToNotificationPage() {
        val intent = Intent(activity, NotificationListActivity::class.java)
        activity.startActivity(intent)
    }
}
